lib/extractor/index.ts

```diff
@@ -8,21 +10,43 @@ import { DockerArchiveManifest, ExtractAction, ExtractedLayers } from "./types";
  * @param extractActions This denotes a file pattern to look for and how to transform the file if it is found.
  * By default the file is returned raw if no processing is desired.
  */
-export async function getDockerArchiveLayersAndManifest(
+export async function getArchiveLayersAndManifest(
+  imageType: ImageType,
   fileSystemPath: string,
   extractActions: ExtractAction[],
-): Promise<{
-  layers: ExtractedLayers;
-  manifest: DockerArchiveManifest;
-}> {
-  const dockerArchive = await extractDockerArchive(
-    fileSystemPath,
-    extractActions,
-  );
+): Promise<ExtractionResult> {
+  switch (imageType) {
+    case ImageType.OciArchive:
+      const ociArchive = await ociExtractor.extractArchive(
+        fileSystemPath,
+        extractActions,
+      );
+
+      return {
+        imageId: ociExtractor.getImageIdFromManifest(ociArchive.manifest),
+        manifestLayers: ociExtractor.getManifestLayers(ociArchive.manifest),
+        extractedLayers: processLayers(ociArchive.layers),
+      };
+    default:
+      const dockerArchive = await dockerExtractor.extractArchive(
+        fileSystemPath,
+        extractActions,
+      );
+
+      return {
+        imageId: dockerExtractor.getImageIdFromManifest(dockerArchive.manifest),
+        manifestLayers: dockerExtractor.getManifestLayers(
+          dockerArchive.manifest,
+        ),
+        extractedLayers: processLayers(dockerArchive.layers),
```

Comment:
Could we name `processLayers` something more specific? Just the name doesn't convey what the processing is! Do we compute something with them? Do we reorder them? If you look at other functions we can tell really easily what they do e.g. `getManifestLayers` or `getImageIdFromManifest`

Reply:
Yeah, you're right :-) With every PR I try to sneak (snyk) one "process" something 😆 
As I can see we basically flatten the array to an object, so I'm going to rename this to `flattenExtractedLayersArray` wdyt?

Reply:
I think flattening is part of it, what it does is it gets the latest modification of a file from all the layers. Maybe `extractedLayersWithLatestFileModifications`? Uhhh maybe it's a bit too long to include in the name (naming is hard 😄)... Or just `layersWithLatestFileModifications`?

Reply:
Naming is very hard!! But I realised I did not quite understand what that function was doing, thanks!! 👍 I updated to `layersWithLatestFileModifications` which makes the most sense I think 😄